core/src/splitting/RegressionSplittingRule.h

```diff
@@ -32,6 +32,42 @@ class RegressionSplittingRule final: public SplittingRule {
 
   ~RegressionSplittingRule();
 
+  /**
+   * Finds the best split at a given node in the tree.
+   *
+   * Is called repeatedly to build a tree in a DFS or BFS fashion.
```

Comment:
I think we only do breadth-first -- I don't see where `find_best_split` is called in a depth-first manner? Also saying breadth-first instead of BFS would be more accurate (there is no 'search' when you're creating a tree).